dataset.py

```diff
@@ -5,135 +5,212 @@
 from numpy.random import MT19937, RandomState, SeedSequence
 from tqdm import tqdm
 
+
 @dataclass
 class PartitionedDataset:
-  train: pd.DataFrame
-  test: pd.DataFrame
-  validation: pd.DataFrame
-
-def partition(df) -> PartitionedDataset:
-  train = df[df["lon"] < -55]
-  test = df[(df["lon"] >= -55) & (df["lat"] > -2.85)]
-  validation = df[(df["lon"] >= -55) & (df["lat"] <= -2.85)]
-  return PartitionedDataset(train, test, validation)
+    '''
```

Comment:
So far, we have used 2 spaces for indent. Is this change to 4 spaces intentional? It is not consistent with the rest of our code or go/pystyle